src/results.cpp

```diff
@@ -945,6 +954,42 @@ REALM_RESULTS_TYPE(util::Optional<double>)
 
 #undef REALM_RESULTS_TYPE
 
+Results Results::freeze(SharedRealm frozen_realm)
+{
+    switch (m_mode) {
+        case Mode::Empty:
+            return Results(true);
+        case Mode::Table:
+            return Results(frozen_realm, *frozen_realm->transaction().import_copy_of(m_table));
+        case Mode::List: {
+            return Results(frozen_realm, frozen_realm->transaction().import_copy_of(*m_list), m_descriptor_ordering);
+        }
+        case Mode::LinkList: {
+            // TODO: Is there an easier/faster way to convert from shared_ptr to unique_ptr? Or should Core API change?
+            LnkLst l = *m_link_list;
+            std::unique_ptr<LnkLst> ull = std::make_unique<LnkLst>(l);
+            std::unique_ptr<LnkLst> frozen_ull = frozen_realm->transaction().import_copy_of(ull);
+            std::shared_ptr<LnkLst> shared_frozen_ull = std::make_shared<LnkLst>(*frozen_ull);
+
+            // If query/sort was provided for the original Results, mode would have changed to Query, so no need
+            // include them here.
+            return Results(frozen_realm, shared_frozen_ull);
+        }
+        case Mode::Query:
+            return Results(frozen_realm, *frozen_realm->transaction().import_copy_of(m_query, PayloadPolicy::Copy), m_descriptor_ordering);
+        case Mode::TableView:
+            auto results = Results(frozen_realm, *frozen_realm->transaction().import_copy_of(m_table_view, PayloadPolicy::Copy), m_descriptor_ordering);
+            results.m_table_view.sync_if_needed();
+            return results;
+    }
+    REALM_COMPILER_HINT_UNREACHABLE();
+}
+
+bool Results::is_frozen()
+{
+    return m_frozen;
```

Comment:
The only reason I introduced the m_frozen member was due to the `Results()` constructor which doesn't take a Realm. I don't know whether it is most meaningful that `Results()` is always considered "frozen" or that it is not?

Reply:
I don't think it's particularly important either way, but I'd be inclined to say that a Results with no Realm should count as frozen because it's not thread-confined and the operations you can perform on it are the same as the ones which can be done on a frozen results.